packages/ui/src/app/pages/Settings/Settings.tsx

```diff
@@ -84,44 +84,29 @@ export const Settings = () => {
 
   const checkRpcEndpoint = async () => {
     // check RPC endpoint
-    try {
+    return await new Promise<boolean>((resolve) => {
       const ws = new WebSocket(customRpcEndpoint)
-      ws.onopen = function () {
-        setIsValidRpcEndpoint(true)
-        return true
-      }
-      ws.onerror = function () {
-        setIsValidRpcEndpoint(false)
-        return false
+      const willResolveTo = (succeeded: boolean, timeout?: any) => () => {
+        if (timeout) clearTimeout(timeout)
+
+        ws.close()
+        setIsValidRpcEndpoint(succeeded)
+        resolve(succeeded)
       }
-    } catch {
-      setIsValidRpcEndpoint(false)
-      return false
-    }
 
-    // timeout
-    await new Promise((res) => setTimeout(res, 3000))
+      const timeout = setTimeout(willResolveTo(false), 3000)
+      ws.onopen = willResolveTo(true, timeout)
+      ws.onerror = willResolveTo(false, timeout)
+    })
   }
 
   const checkQueryEndpoint = async () => {
     // check GraphQL endpoint
     try {
-      const query = `
-        query {
-          electedCouncils{
-            id
-          }
-        }`
-      const response = await fetch(customQueryEndpoint, {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        body: JSON.stringify({ query }),
-      })
-      await response.json()
-      setIsValidQueryEndpoint(true)
-      return true
+      const response = await fetch(customQueryEndpoint + '?query=%7B__typename%7D')
+      const succeeded = response.status < 400 && JSON.parse(await response.json()).data['__typename'] === 'Query'
```

Comment:
Hey, I think this line is failing because `await response.json()` actually returns a js object and not a string, my bad!
```suggestion
      const succeeded = response.status < 400 && (await response.json()).data['__typename'] === 'Query'
```
Since this should be the last change needed, I'll just accept the suggestion myself. Hopefully the formatting is correct :crossed_fingers: